processor/src/lib.rs

```diff
@@ -138,20 +144,25 @@ impl Process<MemAdviceProvider> {
     // CONSTRUCTORS
     // --------------------------------------------------------------------------------------------
     /// Creates a new process with the provided inputs.
-    pub fn new(kernel: &Kernel, inputs: ProgramInputs) -> Self {
-        Self::initialize(kernel, inputs, false)
+    pub fn new(kernel: &Kernel, stack: StackInputs, inputs: ProgramInputs) -> Self {
+        Self::initialize(kernel, stack, inputs, false)
     }
 
     /// Creates a new process with provided inputs and debug options enabled.
-    pub fn new_debug(kernel: &Kernel, inputs: ProgramInputs) -> Self {
-        Self::initialize(kernel, inputs, true)
+    pub fn new_debug(kernel: &Kernel, stack: StackInputs, inputs: ProgramInputs) -> Self {
+        Self::initialize(kernel, stack, inputs, true)
     }
 
-    fn initialize(kernel: &Kernel, inputs: ProgramInputs, in_debug_mode: bool) -> Self {
+    fn initialize(
+        kernel: &Kernel,
+        stack: StackInputs,
+        inputs: ProgramInputs,
+        in_debug_mode: bool,
+    ) -> Self {
```

Comment:
I'd change the signature here to accept the advice provider as input (instead of `ProgramInputs`). So, the signature could be:
```Rust
fn initialize(
    kernel: &Kernel,
    stack_inputs: StackInputs,
    advice_provider: A,
    in_debug_mode: bool,
) -> Self
```

Reply:
Makes sense. Was saving that for the next PR, but we can as well anticipate it here